app/Models/Event.php

```diff
@@ -174,14 +175,32 @@ public function finalize()
         $this->save();
     }
 
-    public function getFulltimeMembersWhereVOG(bool $vog)
+    public function getFulltimeMembersWhereVOG(bool $vog): EloquentCollection
     {
         return $this->members()->wherePivot('wissel', 0)->where('vog', $vog)->get();
     }
 
-    public function getParticipantsWherePaid(bool $paid)
+    public function getParticipantsWherePaid(bool $paid): EloquentCollection
     {
         $operator = $paid ? '>' : '=';
         return $this->participants()->wherePivot('datum_betaling', $operator, '0000-00-00')->get();
     }
+
+    public function training(): ?self
+    {
+        if ($this->type !== 'kamp') {
+            return null;
+        }
```

Comment:
Dit kan eventueel nog strikter door een exceptie te gooien... wat vind jij?

Reply:
Yes, strikter = beter , sowieso in je domein. Op die krijg je een dikke vette exception als een van ons iets niet zo handigs doet , ipv een null die je weer kan negeren.